lib/searches.js

```diff
@@ -129,8 +130,95 @@ var breadthFS = function (url, searchDepth) {
     return searchingPromise;
 }
 
-var depthFS = function () {
-    // To be implemented
+var dfsHelper = function (url, edges, level) {
+    // Get URL and create promise
+    var helperPromise = request({url: url, jar: true});
+    helperPromise.then((body) => {
+        var r = new RegExp('^https?://', 'i');
+        var numInvalidLinks = 0;
+        var randomUrl = "";
+        var isValid = false;
+        
+        // Load HTML and get all links in page
+        var $ = cheerio.load(body);;
+        var sourceTitle = $("title").text();
+        console.log("Link <title>", sourceTitle);
+        links = $('a');
+      
+        // Loop through random page links until an absolute link that looks valid is found.
+        while (!isValid){  
+            if (((typeof(randomUrl) == 'undefined' || !validUrl.isUri(randomUrl)) || !r.test(randomUrl)) && numInvalidLinks < 50) {
```

Comment:
Why do we need to test the url against a Regex? Shouldn't validUrl.isUri return false if it's not valid?

Reply:
For some reason, it doesn't weed everything out.. and I haven't totally figured out what it works with and doesn't yet. If you know of a better library to use to check URLs let me know!